src/libraries/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoder.cs

```diff
@@ -718,7 +734,45 @@ public virtual unsafe int FindFirstCharacterToEncodeUtf8(ReadOnlySpan<byte> utf8
                         // casted to signed byte.
                         int index = Sse2.MoveMask(sourceValue);
 
-                        if (index != 0)
+                        if (index == 0)
+                        {
+                            // All of the following 16 bytes is ASCII.
+
+                            if (Ssse3.IsSupported)
+                            {
+                                Vector128<sbyte> mask = Ssse3Helper.CreateEscapingMask(sourceValue, bitMaskLookupAsciiNeedsEscaping, bitPosLookup, nibbleMaskSByte, nullMaskSByte);
+                                index = Sse2.MoveMask(mask);
+
+                                if (index != 0)
+                                {
+                                    idx += BitHelper.GetIndexOfFirstNeedToEscape(index);
+                                    goto Return;
+                                }
+                            }
+                            else
+                            {
+                                byte* p = (byte*)startingAddress;
+                                if (DoesAsciiNeedEncoding(p[0])) goto Return;
+                                if (DoesAsciiNeedEncoding(p[1])) goto Return1;
```

Comment:
These multiple returns break the dependency-chain on `idx` -- and it's inspired from `SpanHelpers<T>.IndexOf`.

Although I have no strong opinion whether to keep it or not.
With SSSE3 available this won't be executed anyway, but I kept it for other platforms where SSSE3 isn't available.